corehq/apps/repeaters/models.py

```diff
@@ -307,7 +307,7 @@ def post_for_record(self, repeat_record):
             response = simple_post(payload, url, headers=headers, timeout=POST_TIMEOUT, auth=auth)
         except (Timeout, ConnectionError) as error:
             log_repeater_timeout_in_datadog(self.domain)
-            raise RequestConnectionError(error)
+            self.handle_response(RequestConnectionError(error), repeat_record)
```

Comment:
What do you think about this to address your (correct) comment?

The alternative is nesting another try/except in the try block. I'm leaning towards this because it keeps it flat and has the same effect, the only downside being the somewhat surprising choice to pass a newly created exception directly to a function instead of raising it.